pkg/helper/backendapi_entity.go

```diff
@@ -0,0 +1,360 @@
+	}
+	b.resetMappingRules()
+	return nil
+}
+
+func (b *BackendAPIEntity) UpdateMappingRule(id int64, params threescaleapi.Params) error {
+	b.logger.V(1).Info("UpdateMappingRule", "ID", id, "params", params)
+	_, err := b.client.UpdateBackendapiMappingRule(b.backendAPIObj.Element.ID, id, params)
+	if err != nil {
+		return fmt.Errorf("backend [%s] update mappingrule: %w", b.backendAPIObj.Element.SystemName, err)
+	}
+	b.resetMappingRules()
+	return nil
+}
+
+// FindMethodMetricIDBySystemName returns metric or method ID from system name.
+// -1 if metric and method is not found
+func (b *BackendAPIEntity) FindMethodMetricIDBySystemName(systemName string) (int64, error) {
+	metricsMethodList, err := b.MetricsAndMethods()
+	if err != nil {
+		return -1, err
+	}
+
+	for _, metric := range metricsMethodList.Metrics {
+		if metric.Element.SystemName == systemName {
+			return metric.Element.ID, nil
+		}
+	}
+
+	return -1, nil
+}
+
+//
+// PRIVATE
+//
+//
+
+func (b *BackendAPIEntity) resetMethods() {
+	b.metricsAndMethods = nil
+	b.methods = nil
+}
+
+func (b *BackendAPIEntity) resetMetrics() {
+	b.metricsAndMethods = nil
+	b.metrics = nil
+}
+
+func (b *BackendAPIEntity) resetMappingRules() {
+	b.mappingRules = nil
+}
+
+func (b *BackendAPIEntity) getMethods() (*threescaleapi.MethodList, error) {
+	b.logger.V(1).Info("getMethods")
+	hitsID, err := b.getHitsID()
+	if err != nil {
+		return nil, err
+	}
+	methodList, err := b.client.ListBackendapiMethods(b.backendAPIObj.Element.ID, hitsID)
+	if err != nil {
+		return nil, fmt.Errorf("backend [%s] get methods: %w", b.backendAPIObj.Element.SystemName, err)
+	}
+	sanitizeBackendMethodList(methodList)
+
+	return methodList, nil
+}
+
+func (b *BackendAPIEntity) getMetrics() (*threescaleapi.MetricJSONList, error) {
+	b.logger.V(1).Info("getMetrics")
+	metricsAndMethods, err := b.MetricsAndMethods()
+	if err != nil {
+		return nil, err
+	}
+	metricsAndMethodsKeys := make([]string, 0, len(metricsAndMethods.Metrics))
+	for _, metric := range metricsAndMethods.Metrics {
+		metricsAndMethodsKeys = append(metricsAndMethodsKeys, metric.Element.SystemName)
+	}
+
+	methods, err := b.Methods()
+	if err != nil {
+		return nil, err
+	}
+	methodsKeys := make([]string, 0, len(methods.Methods))
+	for _, method := range methods.Methods {
+		methodsKeys = append(methodsKeys, method.Element.SystemName)
+	}
+
+	metricKeys := ArrayStringDifference(metricsAndMethodsKeys, methodsKeys)
+	metricList := &threescaleapi.MetricJSONList{
+		Metrics: make([]threescaleapi.MetricJSON, 0, len(metricKeys)),
+	}
+	for _, systemName := range metricKeys {
+		for _, metric := range metricsAndMethods.Metrics {
+			if metric.Element.SystemName == systemName {
+				metricList.Metrics = append(metricList.Metrics, metric)
+			}
+		}
+	}
+
+	return metricList, nil
+}
+
+func (b *BackendAPIEntity) getMetricsAndMethods() (*threescaleapi.MetricJSONList, error) {
```

Comment:
Why in getMetricsAndMethods we only list metrics?

Reply:
Is it because ListBackendapiMetrics golang Porta client returns metrics AND methods?

Reply:
exactly. 

To get the list of metrics, you need to:
* get the list of metrics and methods
* get the list of methods
* get the difference: (metrics and methods ) - (methods)

Reply:
:shrug:

Reply:
:+1: